app/services/encryptor.rb

```diff
@@ -0,0 +1,15 @@
+class Encryptor
+  ENCRYPTOR = ActiveSupport::MessageEncryptor.new(Rails.application.secrets.secret_key_base[0..31])
+
+  def self.encrypt(text)
+    encrypted_data = ENCRYPTOR.encrypt_and_sign(text)
+
+    CGI.escape(encrypted_data)
```

Comment:
I *think* Rails escapes anything passed into a path helper.

Reply:
It does!

```ruby
[1] pry(main)> Rails.application.routes.url_helpers.candidate_interface_sign_in_url(u: 'example text')
=> "http://localhost:3000/candidate/sign-in?u=example+text"
[2] pry(main)> Rails.application.routes.url_helpers.candidate_interface_sign_in_url(u: 'example text+--')
=> "http://localhost:3000/candidate/sign-in?u=example+text%2B--"
```